openmc/settings.py

```diff
@@ -1043,6 +1061,36 @@ def _create_dd_subelement(self):
             subelement = ET.SubElement(element, "count_interactions")
             subelement.text = str(self._dd_count_interactions).lower()
 
+    def _create_resonance_scattering_element(self):
+        if self.resonance_scattering is None: return
+
+        element = ET.SubElement(self._settings_file, "resonance_scattering")
+
+        # Create an iterable version of resonance_scattering
+        if isinstance(self.resonance_scattering, Iterable):
+            res = self.resonance_scattering
```

Comment:
Why not just make `resonance_scattering` an `Iterable` of `ResonanceScattering` by default? The `resonance_scattering` property could enforce this perhaps.